docs/pages/Configuration.md

```diff
@@ -96,25 +96,144 @@ However, it is recommended that modules include their namespace in the virtual p
 
 ````yaml
 resources:
   configuration:
-    propertiesFileSourceEncoding: UTF-8
+    propertiesFileSourceEncoding: UTF-8|ISO-8859-1
 ````
 
+## Framework Configuration
+
+!!! info
+    This configuration is available since UI5 CLI [`v2.0.0`](https://github.com/SAP/ui5-cli/releases/tag/v2.0.0)
+    and applies only to projects defining [Specification Version](#specification-versions)
+    2.0 or higher.
+
+Define your projects framework dependencies.
+
+Your projects framework configuration you must define whether you want to use the OpenUI5 or the SAPUI5 framework. Also see our [documentation on the differences between OpenUI5 and SAPUI5](./SAPUI5.md#differences-between-openui5-and-sapui5).
+
+**SAPUI5**
+```yaml
+framework:
+  name: SAPUI5
+```
+
+**OpenUI5**
+```yaml
+framework:
+  name: OpenUI5
+```
+
+If you are not sure which framework is right for you, see our [documentation on the differences between OpenUI5 and SAPUI5](./SAPUI5.md#differences-between-openui5-and-sapui5).
+
+!!! warning
+    Projects that use the OpenUI5 framework can not depend on projects that use the SAPUI5 framework.
+
+If you want to execute UI5 CLI commands directly in your project you also need to specify the framework version you want to use. Whenever you execute a UI5 CLI command, the framework version of the current root project is used.
+
+```yaml
+framework:
+  name: SAPUI5
+  version: 1.76.0
+```
+
+You can find an overview of the available versions for each framework here:
+
+- [SAPUI5 Version Overview](http://ui5.sap.com/versionoverview.html) (**Note:** The UI5 Tooling can only consume SAPUI5 starting with version 1.76.0.)
+- [OpenUI5 Version Overview](https://openui5.hana.ondemand.com/versionoverview.html)
+
+### Dependencies
+
+All libraries required by your project should be listed in the libraries section of the framework configuration.
+
+```yaml
+framework:
+  name: SAPUI5
+  version: 1.76.0
+  libraries:
+    - name: sap.ui.core
+    - name: sap.m
+    - name: sap.ui.comp
+    - ...
+```
+
+#### Development Dependencies
+Development dependencies are only installed if the project defining them is the current root project.
+They are typically only required during the development of the project.
+
+```yaml
+  libraries:
+    - name: sap.ushell
+      development: true
+```
+
+Note that a development dependency can not be optional and vice versa.
+
+#### Optional Dependencies
+Optional dependencies are installed either if the project defining them is the current root project or if the dependency is already part current dependency tree. A typical use case is libraries defining optional dependencies to all theme libraries they support.
+The application that is consuming the library can then choose which theme library to use by declaring it as a non-optional dependency.
```

Comment:
You can choose which theme library to use by the application that is consuming the library by declaring it as a non-optional dependency.

Reply:
Done